flink-streaming-java/src/test/java/org/apache/flink/streaming/api/functions/source/FileSplitMonitoringFunctionITCase.java

```diff
@@ -0,0 +1,271 @@
+			Assert.assertEquals(collectedContent.size(), hdPathContents.size());
+			for (Integer fileIdx: collectedContent.keySet()) {
+				Assert.assertEquals(collectedContent.get(fileIdx), hdPathContents.get(fileIdx));
+			}
+			hdPathContents.clear();
+
+			src.cancel();
+			try {
+				src.close();
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+
+		@Override
+		public void invoke(String value) throws Exception {
+			synchronized (lock) {
+				int fileIdx = Character.getNumericValue(value.charAt(0));
+
+				Integer counter = elementCounters.get(fileIdx);
+				if (counter == null) {
+					counter = 0;
+				} else if (counter == LINES_PER_FILE) {
+					// ignore duplicate lines.
+					Assert.fail("Duplicate line: " + value);
+				}
+				elementCounters.put(fileIdx, ++counter);
+
+				String content = collectedContent.get(fileIdx);
+				if (content == null) {
+					content = "";
+				}
+				collectedContent.put(fileIdx, content + value);
+
+				elementCounter++;
+				if (elementCounter == NO_OF_FILES * LINES_PER_FILE) {
+					throw new SuccessException();
+				}
+			}
+		}
+	}
+
+	/**
+	 * A separate thread creating {@link #NO_OF_FILES} files, one file every {@link #INTERVAL} milliseconds.
+	 * It serves for testing the file monitoring functionality of the {@link FileSplitMonitoringFunction}.
+	 * The files are filled with data by the {@link #fillWithData(String, String, int, String)} method.
+	 * */
+	private class FileCreator implements Runnable {
+
+		private final long interval;
+
+		FileCreator(long interval) {
+			this.interval = interval;
+		}
+
+		public void run() {
+			try {
+				for(int i = 0; i < NO_OF_FILES; i++) {
+					synchronized (lock) {
+						fillWithData(hdfsURI, "file", i, "This is test line.");
+					}
+					Thread.sleep(interval);
+				}
+			} catch (IOException e) {
+				e.printStackTrace();
+			} catch (InterruptedException e) {
+				// we just close without any message.
+			}
+		}
+	}
+
+	/**
+	 * Fill the file with content.
+	 * */
+	private void fillWithData(String base, String fileName, int fileIdx, String sampleLine) throws IOException {
+		assert (hdfs != null);
+
+		org.apache.hadoop.fs.Path file = new org.apache.hadoop.fs.Path(base + "/" + fileName + fileIdx);
+
+		org.apache.hadoop.fs.Path tmp = new org.apache.hadoop.fs.Path(base + "/." + fileName + fileIdx);
+		FSDataOutputStream stream = hdfs.create(tmp);
+		StringBuilder str = new StringBuilder();
+		for(int i = 0; i < LINES_PER_FILE; i++) {
+			String line = fileIdx +": "+ sampleLine + " " + i +"\n";
+			str.append(line);
+			stream.write(line.getBytes());
+		}
+		stream.close();
+
+		hdfs.rename(tmp, file);
+
+		hdPathContents.put(fileIdx, str.toString());
+
+		Assert.assertTrue("No result file present", hdfs.exists(file));
+	}
+
+	/**
+	 * The testing {@link FileInputFormat} to be used throughout the tests.
+	 * */
+	private static class StringFileFormat extends FileInputFormat<String> {
```

Comment:
I think this can be replaced by using `TextInputFormat`, both in this ITCase and the Test.